src/workload.rs

```diff
@@ -100,18 +100,18 @@ pub struct GatewayAddress {
 pub mod gatewayaddress {
     use super::{NamespacedHostname, NetworkAddress};
     #[derive(Debug, Hash, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
-    #[serde(tag = "address", content = "content")]
+    #[serde(untagged)]
     pub enum Destination {
-        Hostname(NamespacedHostname),
         Address(NetworkAddress),
+        Hostname(NamespacedHostname),
     }
 }
 
 pub mod address {
     use crate::workload::{Service, Workload};
 
     #[derive(Debug, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
-    #[serde(tag = "address", content = "content")]
+    #[serde(untagged)]
     pub enum Address {
         Workload(Box<Workload>),
```

Comment:
if someone provides a service that just has fields from workload, now it will serialize as the wrong type? can we at least make some types required in proto so that cannot happen

Reply:
im being pretty paranoid but this was a pain to debug the first time